calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseExternalCalendarSubscriptionService.java

```diff
@@ -314,10 +318,36 @@ public Calendar getCalendarSubscription(String reference)
 		m_log.debug("ExternalCalendarSubscriptionService.getCalendarSubscription("
 				+ reference + ")");
 		m_log.debug(" |-> subscriptionUrl: " + subscriptionUrl);
+		
+		String currentSessionUserId = m_sessionManager.getCurrentSessionUserId();
+		
+		//External references will not work for anonymous users, no idea how to track circular references for them.
+		if ((currentSessionUserId==null)||(currentSessionUserId.isEmpty())) return null;
 
-		ExternalSubscription subscription = getExternalSubscription(subscriptionUrl,
-				_ref.getContext());
-
+		ExternalSubscription subscription = null;
+		if (!externalCalRefs.containsKey(currentSessionUserId))
```

Comment:
Doing a `.contains()` followed by a '.get()' is a little wasteful as I recently learned, as you're performing the hash and bucket search twice. Instead, just do a `.get()` outside the loop, and check the return for null in the `if()` rather than doing the contains:

```
String urlList = externalCalRefs.get( currentSessionUserId );
if( urlList == null )
{ ... }
else
{ ... }
```